app/tests/infrastructure/test_exchange_repository.py

```diff
@@ -64,3 +64,34 @@ def test_delete_should_delete_indicated_exchange(self, mock_get_table, an_exchan
 
         # Then
         assert exchange_repository.get(an_exchange.identifier) is None
+
+    @patch('hallebarde.infrastructure.exchange_repository._get_dynamodb_table')
+    def test_get_identifier_from_token_should_return_identifier_associated_with_given_token(self, mock_get_table,
+                                                                                            an_exchange,
+                                                                                            get_dynamodb_table):
+        # Given
+        mock_get_table.return_value = get_dynamodb_table
+        exchange_repository.save(an_exchange)
+
+        # When
+        identifier_from_upload_token = exchange_repository.get_identifier_from_token(upload_token=an_exchange.upload_token)
+        identifier_from_download_token = exchange_repository.get_identifier_from_token(download_token=an_exchange.download_token)
+
+        # Then
+        assert identifier_from_upload_token == an_exchange.identifier
+        assert identifier_from_download_token == an_exchange.identifier
+
+    @patch('hallebarde.infrastructure.exchange_repository._get_dynamodb_table')
+    def test_get_identifier_from_token_should_return_none_if_exchange_does_not_exist(self, mock_get_table,
```

Comment:
Add a NoneSafeness test -> what if up and dl token are None ? should return None then